components/NavLayout/SideBar.js

```diff
@@ -1,12 +1,183 @@
-      <h3>My Classes</h3>
-      <h3>Calendar</h3>
-      <h3>Notifications</h3>
-      <h3>Class Catalog</h3>
-    </nav>
+    <Box sx={{ display: 'flex' }}>
+      <CssBaseline />
+      <AppBar position="fixed" open={open} className={styles.topbar}>
+        <Toolbar>
+          <IconButton
+            color="inherit"
+            aria-label="open drawer"
+            onClick={handleDrawerOpen}
+            edge="start"
+            sx={{
+              marginRight: 5,
+              ...(open && { display: 'none' }),
+            }}
+          >
+            <MenuIcon />
+          </IconButton>
+          <Typography className={styles.siteTitle} variant="h5" noWrap component="div">
+            ClassMates
+          </Typography>
+          <Container className={styles.profileIcon} sx={{ flexGrow: 0 }}>
+            <Typography className={styles.profileName}>Current User</Typography>
+            <Avatar
+              className={styles.profileAvatar}
+              alt="Remy Sharp"
+              src="https://www.vhv.rs/dpng/d/426-4264903_user-avatar-png-picture-avatar-profile-dummy-transparent.png"
+              sx={{ width: 50, height: 50 }}
+            />
+          </Container>
+        </Toolbar>
+      </AppBar>
+      <Drawer className='drawer' variant="permanent" open={open} >
+        <DrawerHeader>
+          <IconButton onClick={handleDrawerClose}>
+            {theme.direction === 'rtl' ? <ChevronRightIcon style={{ color: '#FFFFFF' }} /> : <ChevronLeftIcon style={{ color: '#FFFFFF' }} />}
+          </IconButton>
+        </DrawerHeader>
+
+        <List>
+          {['My Classes', 'Calendar', 'Notifications', 'Class Catalog'].map((text, index) => (
+            <ListItemButton
+              key={text}
+              sx={{
+                minHeight: 48,
+                justifyContent: open ? 'initial' : 'center',
+                px: 2.5,
+              }}
+            >
+              <ListItemIcon
+                sx={{
+                  minWidth: 0,
+                  mr: open ? 3 : 'auto',
+                  justifyContent: 'center',
+                }}
+              >
+                {index === 0 && <Link href={pageUrls[index]}><ClassOutlinedIcon style={{ color: '#FFFFFF' }} /></Link>}
+                {index === 1 && <Link href={pageUrls[index]}><CalendarMonthOutlinedIcon style={{ color: '#FFFFFF' }} /></Link>}
+                {index === 2 && <Link href={pageUrls[index]}><NotificationsOutlinedIcon style={{ color: '#FFFFFF' }} /></Link>}
+                {index === 3 && <Link href={pageUrls[index]}><SearchOutlinedIcon style={{ color: '#FFFFFF' }} /></Link>}
+              </ListItemIcon>
+              <Link href={pageUrls[index]}><ListItemText primary={text} sx={{ opacity: open ? 1 : 0 }} /></Link>
+            </ListItemButton>
+          ))}
+        </List>
+      </Drawer>
+      <Box component="main" sx={{ flexGrow: 1, p: 3 }}>
+        <DrawerHeader />
+        <div className='justWork'>
```

Comment:
Loollllll. Whenever you get a chance adjust the naming. I feel that tho

Reply:
hahaha.... yeeeeah i got you